packages/headless/src/controllers/case-field/headless-case-field.ts

```diff
@@ -112,17 +112,19 @@ export function buildCaseField(
     'buildCaseField'
   ) as Required<CaseFieldOptions>;
 
-  dispatch(
-    registerCaseField({
-      fieldName: options.field,
-      fieldValue: '',
-    })
-  );
-
   const getState = () => {
     return engine.state;
   };
 
+  if (!getState().caseField?.fields?.[options.field]) {
```

Comment:
I'd put this condition in a well name variable like `isRegistered` so we know why this check is there